### PR TITLE
SSL on ELBs

### PR DESCRIPTION
**Pull request by mattmb**
This first: #21

### CODE REVIEW

File: stacks/elb_ssl.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+       [
+          "arn:aws:iam::",
+          {
+             "Ref":"AWS::AccountId"
+          },
+          ":server-certificate/"
+       ]
+    ]
+  }
+}
```
Comment (ashb) [inline stacks/elb_ssl.json]: This file is in the wrong place - you've updated the code to use pkgutil.get_data but not moved the file to live at awsutils/stacks/elb_ssl.json

Comment (mattmb) [inline stacks/elb_ssl.json]: Good spot, thanks :+1:

File: README.md [hunk 2]
```diff
@@ -126,5 +127,23 @@ dev:
...
+        blahblahblah
+        -----END RSA PRIVATE KEY-----
+      chain: |
+        -----BEGIN CERTIFICATE-----
+        blahblahblah
+        -----END CERTIFICATE-----
 ```
 
+##### SSL certs for ELBs
+If you set the protocol on an ELB to HTTPS you must include a key called `certificate_name` and matching cert data in a key with the same name under `ssl` (see example above). The `cert` and `key` are required and the `chain` is optional.
```
Comment (ashb) [inline README.md]: `certificate_name`  given the example YAML shuld be `my-cert` I believe.

We also perhaps should make it clear that we recommend storing the cert in the secrets/passwords YAML, not the main project one.

Comment (ashb) [inline README.md]: Oh actually it would be `my-cert-$stackname` wouldn't it

Comment (mattmb) [inline README.md]: let me update that so it's clearer, we just add that stack-name when we upload/delete not in the config.

File: tests/tests.py [hunk 3]
```diff
@@ -198,7 +198,134 @@ def test_elb(self):
...
+                  'Protocol': 'TCP'
+                },
+                { 'LoadBalancerPort': 443,
+                  'InstancePort': 443,
+                  'Protocol': 'HTTPS'
+                },
+            ],
+        }]
+        config = ConfigParser(project_config.config,'my-stack-name')
+        self.assertRaises(errors.CfnConfigError)
```
Comment (ashb) [inline tests/tests.py]: You aren't actually testing anything here: when called like this it is returning an Context Manager: https://docs.python.org/2.7/library/unittest.html#unittest.TestCase.assertRaises

``` python
with self.assertRaises(errors.CfnConfigError):
  config.ssl()
```

is what you want

File: tests/tests.py [hunk 4]
```diff
@@ -198,7 +198,134 @@ def test_elb(self):
...
```
Comment (ashb) [inline tests/tests.py]: Same here - missing the call

**Comment by ashb** [comment]
Some rebasing and squashing and we should be good to :ship: :it:.

**Comment by mattmb** [comment]
@ashb fixed the tests and squashed it down a bit :smile:

**Pull request merged**